mlir/lib/Dialect/Affine/Analysis/Utils.cpp

```diff
@@ -1804,29 +1804,28 @@ void mlir::affine::getComputationSliceState(
   unsigned pos = isBackwardSlice ? numSrcLoopIVs + loopDepth : loopDepth;
   unsigned num =
       isBackwardSlice ? numDstLoopIVs - loopDepth : numSrcLoopIVs - loopDepth;
-  dependenceConstraints->projectOut(pos, num);
+  FlatAffineValueConstraints sliceCst(dependenceConstraints);
```

Comment:
I have some questions regarding this. Why are you taking a const FlatAffineValueConstraints& as argument and then creating a copy here? If you think that `dependenceConstraints` might change, you can remove the const.
Also, why did you change the signature, i.e. const ref instead of pointer?
The reasoning shouldbe clearly mentioned in the commit summary (no mention of this change currently!)

Reply:
This method shouldn't be updating `dependenceConstraints` within itself - it's meant to be an input to this, and its user doesn't expect/want it to be updated.

Reply:
Strictly speaking, it was incorrect behavior to have been updating 'dependenceConstraints' - this is a `get` method. The doc comment already mentions it as input. I'll update the commit summary to reflect the fix.

Reply:
Why not pass the argument by value then? Anyways you're creating a separate copy of `dependenceConstraints` named `sliceCst` inside the function body.

Reply:
Non-trivial inputs have to be passed const reference, not by value. If the method needs a copy of it for temporary working and modification, it should do it itself. The caller shouldn't be bothered/have to think about it. The copy is being created because `sliceCst` is modified later - it's a scratch constraints.